.github/workflows/deploy.yml

```diff
@@ -0,0 +1,30 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - prod
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set Up node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 20.x
+          cache: "yarn"
+
+      - name: Install dependencies
+        run: yarn --immutable --immutable-cache
+
+      - name: Build
+        run: yarn build
+
+      - name: Deploy gh-pages if on prod
```

Comment:
fwiw I learned we can add dev dependency `gh-pages` and add a deploy command like `yarn gh-pages`